core/src/main/java/org/elasticsearch/common/transport/NetworkExceptionHelper.java

```diff
@@ -35,7 +36,7 @@ public static boolean isConnectException(Throwable e) {
     }
 
     public static boolean isCloseConnectionException(Throwable e) {
-        if (e instanceof ClosedChannelException) {
+        if (e instanceof ClosedChannelException || e instanceof SocketException) {
```

Comment:
>  the closed flag on the channel 

What flag do you mean exactly?

> just look at where that ctor is used there are so many instance of this I think just assuming that the socket is closed is wrong

OK. I'll remove the check. We can always reduce noise if we see too many warn logs.